stripe/src/main/java/com/stripe/android/PaymentConfiguration.java

```diff
@@ -0,0 +1,66 @@
+package com.stripe.android;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import com.stripe.android.model.Address;
+
+public class PaymentConfiguration {
+
+    private static PaymentConfiguration mInstance;
+
+    private @NonNull String mPublishableKey;
+    private @Address.RequiredAddressFields int mRequiredAddressFields;
+    private boolean mShouldUseSources;
+
+    private PaymentConfiguration(@NonNull String publishableKey) {
+        mPublishableKey = publishableKey;
+    }
+
+    @NonNull
+    public static PaymentConfiguration getInstance() {
+        if (mInstance == null) {
+            throw new RuntimeException(
+                    "Attempted to get instance of PaymentConfiguration without initialization.");
+        }
+        return mInstance;
+    }
+
+    public static void init(@NonNull String publishableKey) {
+        mInstance = new PaymentConfiguration(publishableKey);
+        mInstance.mRequiredAddressFields = Address.RequiredAddressFields.NONE;
+        mInstance.mShouldUseSources = true;
+    }
+
+    @NonNull
+    public String getPublishableKey() {
+        return mPublishableKey;
+    }
+
+    public @Address.RequiredAddressFields int getRequiredAddressFields() {
+        return mRequiredAddressFields;
+    }
+
+    @NonNull
+    public PaymentConfiguration setRequiredAddressFields(
+            @Address.RequiredAddressFields int requiredAddressFields) {
+        mRequiredAddressFields = requiredAddressFields;
+        return this;
+    }
+
+    public boolean getShouldUseSources() {
```

Comment:
We settled on `useSourcesForCards` on iOS. This switch only controls whether we turn card into source or tokens, and I think it's good to be explicit – other payment methods will always use sources. cc @bdorfman-stripe 
https://github.com/stripe/stripe-ios/blob/sources-ui/Stripe/PublicHeaders/STPPaymentConfiguration.h#L75